django/contrib/gis/db/backends/postgis/adapter.py

```diff
@@ -48,24 +48,27 @@ def __str__(self):
     def _fix_polygon(cls, poly):
         return poly
 
-    def prepare(self, conn):
-        """
-        This method allows escaping the binary in the style required by the
-        server's `standard_conforming_string` setting.
-        """
-        if self.is_geometry:
-            self._adapter.prepare(conn)
-
     def getquoted(self):
         """
         Return a properly quoted string for use in PostgreSQL/PostGIS.
         """
+        if PSYCOPG_VERSION >= (3, 0, 0):
+
+            def quote(obj):
+                return sql.quote(obj).encode()
+
+        else:
+            from psycopg2.extensions import adapt
+
+            def quote(obj):
+                return adapt(obj).getquoted()
+
```

Comment:
This can be moved to a class level definition instead based of `PSYCOPG_VERSION`